# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [29]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I chose the Content Refresh lane because I want to understand which existing pages may need improvement based on their search performance. This is useful because pages that already have visibility may lose performance over time, and identifying those pages can help a team decide where to focus its content-refresh effort first. I chose this lane because it connects machine learning with a practical decision: deciding which pages should be reviewed for possible improvement.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [30]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The decision is which existing pages should be prioritized for a content-refresh review. A content or SEO team would use the recommendation to decide which pages to investigate and improve first. A wrong recommendation could waste the team's time and effort on a page that does not need attention, while a page that actually needs review could be missed. Therefore, the goal is to rank pages for review rather than claim that the model can guarantee which page will improve.


In [31]:
import pandas as pd

url = "https://raw.githubusercontent.com/vk-7524/flyrank-ml-intrnship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))
print("Columns:", list(df.columns))

Number of rows: 30000
Number of columns: 44
Columns: ['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [32]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


In [33]:
import pandas as pd

url = "https://raw.githubusercontent.com/vk-7524/flyrank-ml-intrnship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of pages:", len(df))
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average clicks (90d):", round(df["clicks_90d"].mean(), 2))

Number of pages: 30000
Average impressions (90d): 5200.37
Average clicks (90d): 16.1


The starter dataset contains 30,000 pages. The average number of impressions over 90 days is 5,200.37, and the average number of clicks over 90 days is 16.10. These numbers show that the dataset contains measurable search-performance information that can be explored to identify pages that may deserve content-refresh review.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [34]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


My work can report observed and measured patterns in the starter data and provide directional, decision-support recommendations about which pages may deserve further review. It cannot prove that one factor causes a change in search performance, and it cannot claim to predict or explain Google's ranking algorithm. The results should be treated as evidence for prioritization and further investigation, not as causal proof or a guarantee of future performance.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.